controllers/metricstorage_controller.go

```diff
@@ -510,6 +518,60 @@ func (r *MetricStorageReconciler) reconcileNormal(
 	return ctrl.Result{}, nil
 }
 
+// PrometheusEndpointSecret creates a Secret that contains the details for Prometheus API endpoint
+func (r *MetricStorageReconciler) prometheusEndpointSecret(
+	ctx context.Context,
+	instance *telemetryv1.MetricStorage,
+	helper *helper.Helper,
+	labels map[string]string,
+) error {
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-prometheus-endpoint", instance.Name),
+			Namespace: instance.Namespace,
+			Labels:    labels,
+		},
+	}
+
+	secret.Data = map[string][]byte{
+		"host": []byte(fmt.Sprintf("%s-prometheus.%s.svc", telemetryv1.DefaultServiceName, instance.Namespace)),
+		"port": []byte(strconv.Itoa(telemetryv1.DefaultPrometheusPort)),
+	}
+
+	if _, err := controllerutil.CreateOrUpdate(context.TODO(), helper.GetClient(), secret, func() error {
+		secret.Type = corev1.SecretTypeOpaque
+
+		err := controllerutil.SetControllerReference(instance, secret, helper.GetScheme())
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}); err != nil {
+		return err
+	}
+
+	if instance.Spec.PrometheusTLS.Enabled() {
+		tlsSecret := &corev1.Secret{
+			Data: map[string][]byte{
+				"ca_secret": []byte(fmt.Sprintf("cert-%s-prometheus-svc", telemetryv1.DefaultServiceName)),
+				"ca_key":    []byte("ca.crt"),
```

Comment:
We shouldn't assume the name of the secret. AFAIK this can be configured by users somehow instead of using the ones generated by O-O.
```suggestion
				"ca_secret": []byte(instance.Spec.PrometheusTLS.SecretName),
				"ca_key":    []byte(tls.CaKey),
```

Reply:
See https://github.com/openstack-k8s-operators/telemetry-operator/blob/main/pkg/metricstorage/prometheus_tls_patch.go#L53 for what's passed to Prometheus